pkg/linode-bs/luks.go

```diff
@@ -63,31 +64,21 @@ func (ctx *LuksContext) validate() error {
 		return nil
 	}
 
-	var appendFn = func(x string, xs string) string {
-		if xs != "" {
-			xs += "; "
-		}
-		xs += x
-		return xs
-	}
-
-	errorMsg := ""
+	var err error
 	if ctx.VolumeName == "" {
-		errorMsg = appendFn("no volume name provided", errorMsg)
+		err = errors.Join(err, fmt.Errorf("no volume name provided"))
 	}
 	if ctx.EncryptionKey == "" {
-		errorMsg = appendFn("no encryption key provided", errorMsg)
+		err = errors.Join(err, fmt.Errorf("no encryption key provided"))
 	}
 	if ctx.EncryptionCipher == "" {
-		errorMsg = appendFn("no encryption cipher provided", errorMsg)
+		err = errors.Join(err, fmt.Errorf("no encryption cipher provided"))
 	}
 	if ctx.EncryptionKeySize == "" {
-		errorMsg = appendFn("no encryption key size provided", errorMsg)
+		err = errors.Join(err, fmt.Errorf("no encryption key size provided"))
 	}
```

Comment:
should `fmt.Errorf`s be `errors.New` instead since we're not formatting the error message?

Reply:
Yeah I agree that they should be. I don't know why I just always reach for `fmt.Errorf`.